src/App.js

```diff
@@ -1,16 +1,41 @@
 import React from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+// import ChatEntry from './components/ChatEntry';
+import ChatLog from './components/ChatLog';
+import {useState} from 'react'
 
 const App = () => {
+  
+  const[textMessage, setTextMessage] = useState(chatMessages)
+  const likedMessage = (id) => {
+  const isliked = textMessage.map((text) =>{
+    if (text.id === id){
+      return {...text, liked: !text.liked}
+    } else{
+      return text;
+    }
+  })
+    setTextMessage(isliked);
+  }
+
+  const totalLikes = (data)=> {
+    const newList = data.filter((ele)=>ele.liked===true)
+    return newList.length
```

Comment:
Filter usually implies that we are trying to filter something out by a condition. Prefer to use `reduce`, which will achieve the same thing, but is more descriptive of what we're trying to do which is to take many values and reduce it down to one summed value. Like:
```javascript
  // no need to pass in an argument called `data` since `textMessage` is already in scope
  const totalLikes = () => { 
    return textMessage.reduce((total, text) => {
      return text.liked ? total + 1 : total;
    }, 0)
  };